spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -933,6 +949,13 @@ struct ProductionVisitor
 
         pb->finalizeUnit(true, p.location());
 
+        if ( auto a = AttributeSet::find(p.unitType().attributes(), "&max-size") ) {
+            // Restore parser state.
+            auto ncur = state().ncur;
+            popState();
+            builder()->addAssign(state().cur, *ncur);
+        }
+
```

Comment:
As we just talked about, this isn't reliably triggering an error in all cases, let's trigger it here directly.